src/core/mac/sub_mac.cpp

```diff
@@ -985,14 +1005,14 @@ void SubMac::HandleCslTimer(void)
         mCslState = kCslSample;
 
         mCslSampleTime += mCslPeriod * kUsPerTenSymbols;
-        Get<Radio>().UpdateCslSampleTime(mCslSampleTime.GetValue());
-
-        mCslTimer.StartAt(mCslSampleTime, kCslSampleWindow);
+        duration = GetCslSampleDuration();
+        mCslTimer.StartAt(mCslSampleTime, duration);
```

Comment:
As `mCslSampleTime` is at the center of CSL sample window, the end of sample window should be sampleTime + drift + uncertainty + `kMinCslWindow`. Not need to X2 in `GetCslSampleDuration` for drift and `mCslParentUncert`?